zendesk/resource_zendesk_group_test.go

```diff
@@ -63,25 +124,82 @@ func TestCreateGroup(t *testing.T) {
 	}
 }
 
-func TestUnmarshalGroup(t *testing.T) {
-	m := &identifiableMapGetterSetter{
-		id: "1234",
-		mapGetterSetter: mapGetterSetter{
-			"url":  "https://example.zendesk.com/api/v2/ticket_fields/360011737434.json",
-			"name": "name",
-		},
+func TestUpdateGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id:              "12345",
+		mapGetterSetter: make(mapGetterSetter),
 	}
 
-	g, err := unmarshalGroup(m)
-	if err != nil {
-		t.Fatalf("Could marshal map %v", err)
+	m.EXPECT().UpdateGroup(Eq(int64(12345)), Any()).Return(zendesk.Group{}, nil)
+	if err := updateGroup(i, m); err != nil {
+		t.Fatal("readGroup returned an error")
 	}
+}
 
-	if v := m.Get("url"); g.URL != v {
-		t.Fatalf("group had url value %v. should have been %v", g.URL, v)
+func TestDeleteGroup(t *testing.T) {
+	ctrl := NewController(t)
+	defer ctrl.Finish()
+
+	m := mock.NewClient(ctrl)
+	i := &identifiableMapGetterSetter{
+		id: "12345",
 	}
 
-	if v := m.Get("name"); g.Name != v {
-		t.Fatalf("group had name value %v. should have been %v", g.Name, v)
+	m.EXPECT().DeleteGroup(Eq(int64(12345))).Return(nil)
+	if err := deleteGroup(i, m); err != nil {
+		t.Fatal("readGroup returned an error")
```

Comment:
`delete group returned an error` also should probably write the error value to the message.